internal/stack/serverless.go

```diff
@@ -0,0 +1,462 @@
+		elasticsearch.OptionWithPassword(project.Credentials.Password),
+	)
+	if err != nil {
+		return project, fmt.Errorf("failed to create elasticsearch client")
+	}
+
+	project.KibanaClient, err = NewKibanaClient(
+		kibana.Address(project.Endpoints.Kibana),
+		kibana.Username(project.Credentials.Username),
+		kibana.Password(project.Credentials.Password),
+	)
+	if err != nil {
+		return project, fmt.Errorf("failed to create kibana client")
+	}
+
+	return project, nil
+}
+
+func (sp *serverlessProvider) deleteProject(project *serverless.Project, options Options) error {
+	return sp.client.DeleteProject(project)
+}
+
+func (sp *serverlessProvider) currentProject(config Config) (*serverless.Project, error) {
+	projectID, found := config.Parameters[paramServerlessProjectID]
+	if !found {
+		return nil, errProjectNotExist
+	}
+
+	projectType, found := config.Parameters[paramServerlessProjectType]
+	if !found {
+		return nil, errProjectNotExist
+	}
+
+	project, err := sp.client.GetProject(projectType, projectID)
+	if err == serverless.ErrProjectNotExist {
+		return nil, errProjectNotExist
+	}
+	if err != nil {
+		return nil, fmt.Errorf("couldn't check project health: %w", err)
+	}
+
+	project.Credentials.Username = config.ElasticsearchUsername
+	project.Credentials.Password = config.ElasticsearchPassword
+
+	project, err = sp.createClients(project)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create project client")
+	}
+
+	fleetURL := config.Parameters[paramServerlessFleetURL]
+	if true {
+		fleetURL, err = project.DefaultFleetServerURL()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get fleet URL: %w", err)
+		}
+	}
+	project.Endpoints.Fleet = fleetURL
+
+	return project, nil
+}
+
+func getProjectSettings(options Options) (projectSettings, error) {
+	s := projectSettings{
+		Name:         createProjectName(options),
+		Type:         options.Profile.Config(configProjectType, defaultProjectType),
+		Region:       options.Profile.Config(configRegion, defaultRegion),
+		StackVersion: options.StackVersion,
+	}
+
+	return s, nil
+}
+
+func createProjectName(options Options) string {
+	return fmt.Sprintf("elastic-package-test-%s", options.Profile.ProfileName)
+}
+
+func newServerlessProvider(profile *profile.Profile) (*serverlessProvider, error) {
+	client, err := serverless.NewClient()
+	if err != nil {
+		return nil, fmt.Errorf("can't create serverless provider: %w", err)
+	}
+
+	return &serverlessProvider{profile, client}, nil
+}
+
+func (sp *serverlessProvider) BootUp(options Options) error {
+	logger.Warn("Elastic Serverless provider is in technical preview")
+
+	config, err := LoadConfig(sp.profile)
+	if err != nil {
+		return fmt.Errorf("failed to load configuration: %w", err)
+	}
+
+	settings, err := getProjectSettings(options)
+	if err != nil {
+		return err
+	}
+
+	if settings.Type == "elasticsearch" {
+		return fmt.Errorf("serverless project type not supported: %s", settings.Type)
+	}
```

Comment:
Added an error (as not supported) in case an elasticsearch project is being requested.

Reply:
It would be probably better to check for the expected types (`security` and `observability` now), and fail on everything else.